File: 215-kth-largest-element-in-an-array/215-kth-largest-element-in-an-array.java
class Solution {
    public int findKthLargest(int[] nums, int k) {
        int n=nums.length;
        
        PriorityQueue<Integer> pq=new PriorityQueue<>(Collections.reverseOrder());
        
        for(int i=0; i<n; i++){
            pq.add(nums[i]);
        }
        
        int ans=-1;
        
        while(k>0){
            ans=pq.remove();
            k--;
        }
        
        return ans;
    }
}

File: 215-kth-largest-element-in-an-array/215-kth-largest-element-in-an-array.cpp
class Solution {
public:
    int findKthLargest(vector<int>& nums, int k) {
        // nlog(k);
        priority_queue<int,vector<int>,greater<int>> pq;
        
        for(int e:nums){
            pq.push(e);
            
            if(pq.size()>k){
                pq.pop();
            }
        }
        
        return pq.top();
    }
};